app/src/main/java/me/quadphase/qpdex/pokemon/Type.java

```diff
@@ -54,10 +56,10 @@ public String getName() {
      * @param defendingType type that this type is defending against
      * @return the attacking effectiveness against the defendingType
      */
-//    public short getAttackingEffectivenessAgainst (Type defendingType) {
-//        PokemonFactory pokemonFactory = PokemonFactory.getPokemonFactory(null);
-//        return pokemonFactory.getTypeEffectivenessTable()[pokemonFactory.getTypeID(this)][pokemonFactory.getTypeID(defendingType)];
-//    }
+    public double getAttackingEffectivenessAgainst (Type defendingType) {
+        PokemonFactory pokemonFactory = PokemonFactory.getPokemonFactory(null);
+        return pokemonFactory.getTypeEffectivenessTable()[this.typeID - 1][defendingType.getTypeID() - 1];
```

Comment:
Careful here. You can make the sparse array directly index the types rather than having n-1. That will also avoid any sort of exceptional behaviour.